backend/CircleForms/Controllers/Authorization/OAuth/OAuthController.cs

```diff
@@ -140,19 +139,29 @@ public async Task<IActionResult> CompleteAuthentication()
             new(ClaimTypes.Role, "User")
         };
 
-        if (user.Roles.HasFlag(Roles.SuperAdmin))
+        if (_superAdminsId.Contains(userId))
         {
             claims.Add(new Claim(ClaimTypes.Role, "SuperAdmin"));
-        }
-
-        if (user.Roles.HasFlag(Roles.Admin))
-        {
             claims.Add(new Claim(ClaimTypes.Role, "Admin"));
+            claims.Add(new Claim(ClaimTypes.Role, "Moderator"));
         }
-
-        if (user.Roles.HasFlag(Roles.Moderator))
+        else
         {
-            claims.Add(new Claim(ClaimTypes.Role, "Moderator"));
+            //TODO: rewrite it
+            if (user.Roles.HasFlag(Roles.SuperAdmin))
```

Comment:
Superadmins can't at the time have role and not be in the list? Escalating someone to superadmins shouldn't be possible